mc-sema/bin_descend/get_cfg.py

```diff
@@ -1086,10 +1112,142 @@ def processDataChunk(ea, size):
         DEBUG("Found an unknown blob at {:x}, treating as table\n".format(ea))
         return processTable(ea, size)
 
+#referenced from
+# http://stackoverflow.com/questions/32030412/twos-complement-sign-extension-python
+def sign_extend(value, bits):
+    sign_bit = 1 << (bits - 1)
+    return (value & (sign_bit - 1)) - (value & sign_bit)
+
+def checkIfOffsetTable(ea):
+    """
+    Check if this is an offset table: that is, a table
+    off offsets that when added to table base result
+    in the VA of a jump target
+    """
+
+    DEBUG("LOOKING FOR OFFSET TABLE AT: {:08x}\n".format(ea))
+    # preconditions
+    # can only really do this when all sections are 
+    # correctly based
+    if not isLinkedElf():
+        DEBUG("\t... not a linked elf\n");
+        return False, 0, []
+
+    # only present in PIE executables
+    if not PIE_MODE:
+        DEBUG("\t... not in PIE mode\n");
+        return False
+
+    #TODO: revisit
+    if getBitness() != 64:
+        DEBUG("\t... not 64-bit\n");
+        return False, 0, []
+
+    # check that there is a code reference
+    # to ea somewhere
+    refs = [x for x in idautils.DataRefsTo(ea)]
+    if len(refs) == 0:
+        DEBUG("\t... no refs to ea\n");
+        return False, 0, []
+    
+    # assumes 32-bit offsets
+    # 1: EA + EA[n] = beginning of an instruction
+    entrycount = 0
+    entries = []
+    while True:
+        entry = readDword(ea+entrycount*4)
+        # no null entries
+        if entry == 0:
+            break
+
+        dest_guess = ea + sign_extend(entry, 64)
+        dest_guess &= 0xFFFFFFFFL
```

Comment:
Why is this masking right? Is it that, within IDA, you will only ever see things as 32-bit "addresses" because they're really offsets? Is it possible for `ea` to not fit within 32 bits?